client/src/components/Draftail/CommentableEditor/CommentableEditor.js

```diff
@@ -0,0 +1,217 @@
+      undoStack: editorState.getUndoStack(),
+      redoStack: editorState.getRedoStack(),
+    },
+  );
+  return EditorState.acceptSelection(state, state.getSelection())
+};
+
+function CommentableEditor({commentApp, fieldNode, contentPath, rawContentState, onSave, inlineStyles, editorRef, ...options}) {
+    const [editorState, setEditorState] = useState(() => createEditorStateFromRaw(rawContentState));
+    const CommentControl = useMemo(() => getCommentControl(commentApp, contentPath, fieldNode), [commentApp, contentPath, fieldNode]);
+    const commentsSelector = useMemo(() => commentApp.utils.selectCommentsForContentPathFactory(contentPath), [contentPath, commentApp]);
+    const CommentDecorator = useMemo(() => getCommentDecorator(commentApp), [commentApp])
+    const comments = useSelector(commentsSelector, shallowEqual);
+    const enabled = useSelector(commentApp.selectors.selectEnabled);
+    const focusedId = useSelector(commentApp.selectors.selectFocused);
+
+    const ids = useMemo(() => comments.map(comment => comment.localId), [comments]);
+
+    const commentStyles = useMemo(() => ids.map(id => {return {
+      type: COMMENT_STYLE_IDENTIFIER + id,
+      style: enabled ? {
+        'background-color': (focusedId !== id) ? '#01afb0' : '#007d7e'
+      } : {}
+    }}), [ids, enabled]);
+
+    const [uniqueStyleId, setUniqueStyleId] = useState(0)
+
+    const previousFocused = usePrevious(focusedId);
+    const previousIds = usePrevious(ids);
+    const previousEnabled = usePrevious(enabled);
+    useEffect(() => {
+      // Only trigger a focus-related rerender if the current focused comment is inside the field, or the previous one was
+      const validFocusChange = previousFocused !== focusedId && (previousIds && previousIds.includes(previousFocused) || ids.includes(focusedId))
+
+      if ((!validFocusChange) && previousIds === ids && previousEnabled === enabled) {
+        return
+      }
+
+      // Filter out any invalid styles - deleted comments, or now unneeded STYLE_RERENDER forcing styles
+      const filteredContent = filterInlineStyles(inlineStyles.map(style => style.type).concat(ids.map(id => COMMENT_STYLE_IDENTIFIER + id)), editorState.getCurrentContent())
+      // Force reset the editor state to ensure redecoration, and apply a new (blank) inline style to force inline style rerender
+      // This must be entirely new for the rerender to trigger, hence the unique style id, as with the undo stack we cannot guarantee
+      // that a previous style won't persist without filtering everywhere, which seems a bit too heavyweight
+      // This hack can be removed when draft-js triggers inline style rerender on props change
+      setEditorState(editorState => forceResetEditorState(editorState, 
+        Modifier.applyInlineStyle(
+          filteredContent,
+          getFullSelectionState(filteredContent),
+          'STYLE_RERENDER_'+uniqueStyleId
+        )
+      ))
+      setUniqueStyleId((id) => (id + 1) % 200);
```

Comment:
Why modulo 200?

Reply:
This is pretty arbitrary, I just didn't want the style numbers to get too big if you stayed on the page for a while. To trigger the rerender, the added style can't be the same as one that was just filtered out, but there's no other constraints.